### PR TITLE
P value augmented dickey fuller

### PR DESCRIPTION
**Pull request by MaxBenChrist**
I added the pvalue and the used lag as features. 

they were calculated anyway so we get more features for free. 

To implement this I had to change the type of augmented_dickey_fuller from simple to combiner

### CODE REVIEW

**Comment by MaxBenChrist** [comment]
Well, that is weird. locally, the tests are passing, on travis, the build fails. 

I will check if there is a problem with the stats model package version

**Comment by jneuff** [comment]
The relevant test also fails on my machine (linux).

**Comment by MaxBenChrist** [comment]
That is weird. On my Mac it passes.

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ install:
 - pip install .
 - pip install -r requirements.txt
 - pip install -r test-requirements.txt
+- pip freeze
```
Comment (nils-braun) [inline .travis.yml]: Is this somehow related to this PR?

File: tsfresh/feature_extraction/feature_calculators.py [hunk 2]
```diff
@@ -289,26 +289,32 @@ def mean_autocorrelation(x):
...
+        res = np.NaN, np.NaN, np.NaN
     except ValueError:  # occurs if sample size is too small
-        return np.NaN
+        res = np.NaN, np.NaN, np.NaN
+
+    return [("attr_{}".format(config["attr"]),
+                  res[0] if config["attr"] == "teststat"
+             else res[1] if config["attr"] == "pvalue"
+             else res[2] if config["attr"] == "usedlag" else np.NaN)
+            for config in param]
```
Comment (nils-braun) [inline tsfresh/feature_extraction/feature_calculators.py]: Isn't this simpler with a dict or a list? This seems a bit "ugly" :-D

File: tsfresh/feature_extraction/feature_calculators.py [hunk 3]
```diff
@@ -289,26 +289,32 @@ def mean_autocorrelation(x):
...
 
-@set_property("fctype", "simple")
-def augmented_dickey_fuller(x):
+@set_property("fctype", "combiner")
+def augmented_dickey_fuller(x, param):
     """
     The Augmented Dickey-Fuller test is a hypothesis test which checks whether a unit root is present in a time
     series sample. This feature calculator returns the value of the respective test statistic.
 
     See the statsmodels implementation for references and more details.
```
Comment (nils-braun) [inline tsfresh/feature_extraction/feature_calculators.py]: where is the docu of x?

Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: went missing :D, will add it

File: .travis.yml (hunk 1 above)
Comment (MaxBenChrist) [inline .travis.yml]: yes. somehow, the unit tests were locally passing but not on travis

I use the pip freeze at the end of the build to inspect which versions of the packages are installed

Comment (nils-braun) [inline .travis.yml]: Well, ok. But still, you can delete this now, can't you? :-D

Comment (MaxBenChrist) [inline .travis.yml]: I would leave it for future inspections . in my opinion it improves the travis log.

File: tsfresh/feature_extraction/feature_calculators.py (hunk 2 above)
Comment (MaxBenChrist) [inline tsfresh/feature_extraction/feature_calculators.py]: well, the dictionary does not cover the last else part

**Comment by MaxBenChrist** [comment]
Can you life with both the `pip freeze` and the `else` construction @nils-braun ?

**Comment by nils-braun** [comment]
Yep, I'll fix it after the release
